contracts/signature/Eip712Signature.sol

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.8.0 <0.9.0;
+
+abstract contract EIP712Signature {
+    uint256 private nonce;
+
+    function eip712Nonce() public view returns (uint256 signed) {
+        return nonce;
+    }
+
+    function eip712BumpNonce() internal {
+        nonce = nonce + 1;
+    }
+
+    function eip712SignedBy() internal returns (address signer) {
+        if (msg.data.length >= 4 + 65) {
+            (
+                bytes calldata dataTrimmed,
+                uint8 v,
+                bytes32 r,
+                bytes32 s
+            ) = _splitSignature(msg.data);
+
+            bytes32 txHash = _transactionHash(msg.value, dataTrimmed, nonce);
```

Comment:
Why is `msg.value` used for the hash?

Seems like the hash should include `address(this)`, `block.chainid()`, `dataTrimmed`, and `nonce`.

Reply:
Should we exclude value this? Arguably this gives more control to the signer over how the call pattern happens - I guess

Reply:
I don't think the signer should define the relayer's message value.

Reply:
Sorted